document_stores/elasticsearch/tests/test_document_store.py

```diff
@@ -0,0 +1,206 @@
+        docstore.write_documents(
+            [
+                Document(content="Haskell is a functional programming language"),
+                Document(content="Lisp is a functional programming language"),
+                Document(content="Exilir is a functional programming language"),
+                Document(content="F# is a functional programming language"),
+                Document(content="C# is a functional programming language"),
+                Document(content="C++ is an object oriented programming language"),
+                Document(content="Dart is an object oriented programming language"),
+                Document(content="Go is an object oriented programming language"),
+                Document(content="Python is a object oriented programming language"),
+                Document(content="Ruby is a object oriented programming language"),
+                Document(content="PHP is a object oriented programming language"),
+            ]
+        )
+
+        res = docstore._bm25_retrieval("functional", top_k=3)
+        assert len(res) == 3
+        assert "functional" in res[0].content
+        assert "functional" in res[1].content
+        assert "functional" in res[2].content
+
+    def test_write_duplicate_fail(self, docstore: ElasticsearchDocumentStore):
+        """
+        Verify `DuplicateDocumentError` is raised when trying to write duplicate files.
+
+        `DocumentStoreBaseTests` declares this test but we override it since we return
+        a different error message that it expects.
+        """
+        doc = Document(content="test doc")
+        docstore.write_documents([doc])
+        with pytest.raises(DuplicateDocumentError):
+            docstore.write_documents(documents=[doc], policy=DuplicatePolicy.FAIL)
+        assert docstore.filter_documents(filters={"id": doc.id}) == [doc]
+
+    def test_delete_not_empty(self, docstore: ElasticsearchDocumentStore):
+        """
+        Verifies delete properly deletes specified document.
+
+        `DocumentStoreBaseTests` declares this test but we override it since we
+        want `delete_documents` to be idempotent.
+        """
+        doc = Document(text="test doc")
+        docstore.write_documents([doc])
+
+        docstore.delete_documents([doc.id])
+
+        res = docstore.filter_documents(filters={"id": doc.id})
+        assert res == []
+
+    def test_delete_empty(self, docstore: ElasticsearchDocumentStore):
+        """
+        Verifies delete doesn't raises when trying to delete a non-existing document.
+
+        `DocumentStoreBaseTests` declares this test but we override it since we
+        want `delete_documents` to be idempotent.
+        """
+        docstore.delete_documents(["test"])
+
+    def test_delete_not_empty_nonexisting(self, docstore: ElasticsearchDocumentStore):
+        """
+        Verifies delete properly deletes specified document in DocumentStore containing
+        multiple documents.
```

Comment:
I don't understand the docstring in relation to this test.

Reply:
It's a bad docstring I agree, I copied it from `DocumentStoreBaseTests` only to override that test. 

I'll remove it, those tests are going to be reworked in any case.